Lesson16/sahra/script.js

```diff
@@ -0,0 +1,259 @@
+function year(number) {
+  if(number % 4 === 0) {
+      if (number % 100 !== 0) {
+      return 'leap year';
+    } else {
+      return 'not a leap year';
+    }
+} else if(number % 400 === 0){
+  return 'leap year';
+} else if(number % 400 !== 0){
+  return 'not a leap year';
+}}
+console.log(year(2345));
+console.log(year(2004));
+console.log(year(4400));
+
+
+// 7. Write a function that takes in a temperature and checks if it's above or below freezing using if else statement. (H)
+function temperature(number2){
+  if(number2 > 0){
+    return 'above freezing';
+  } else if(number2 < 0){
+    return 'below freezing';
+  }
+}
+console.log(temperature(-22));
+console.log(temperature(35));
+
+// 8.(H) Write a function that takes in a person's height and weight and checks if they are considered underweight, normal, overweight, or obese using if else statement.
+// Body Mass Index (BMI) is a person’s weight in kilograms (or pounds) divided by the square of height in meters (or feet). A high BMI can indicate high body fatness.
+//Formula: weight (kg) / [height (m)]2
+// Below 18.5	Underweight
+// 18.5 – 24.9	Healthy Weight
+// 25.0 – 29.9	Overweight
+// 30.0 and Above	Obesity
+function bodyMassIndex(weight,height){
+ 
+ 
+  if (weight / (height * height )<= 18.5){
+    return 'Underweight';
+  } else if (weight / (height * height) > 18.5 && weight / (height * height) <=24.9 ){
+    return 'Healty Weight';
+  } else if (weight / (height * height) >= 25.0 && weight / (height * height) <=29.9 ){
+    return 'Overweight';
+  } else if (weight / (height * height) >=30.0 ){
+    return 'Above Obesity';
+  } 
+}
+
+console.log(bodyMassIndex(50,1.80));
+console.log(bodyMassIndex(80,1.80));
+console.log(bodyMassIndex(90,1.80));
+console.log(bodyMassIndex(100,1.80));
+
+// 9. Write a function that takes in a person's age and checks if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), or senior citizen (age 65 and above) using if else statement. (H)
+function age(number3){
+  if(number3 >= 0 && number3 <=12){
+    return 'child';
+  } else if (number3 >= 13 && number3 <=19){
+    return 'teenager';
+  } else if (number3 >= 20 && number3 <=64){
+    return 'adult';
+  } else if (number3 >= 65){
+    return 'senior citizen';
+  } 
+}
+console.log(age(5));
+console.log(age(17));
+console.log(age(49));
+console.log(age(80));
+
+// 10. Write a function that takes age as a number and whether they are a student (yes or no). If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". If the age is greater than 65, print "You are a senior citizen". If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote". (H)
+function person(age,student){
+  if( student === 'yes'){
+      if(age < 18){
```

Comment:
In this function, try to use && operator to simplify nested if else statements. E.g.:
```
if (age < 18 && isStudent) {
...
}
```